mbbsd/bbs.c

```diff
@@ -433,6 +433,24 @@ int IsFreeBoardName(const char *brdname)
     return 0;
 }
 
+/*
+ * Check if user ip address is from Taiwan
+ *
+ * This function will compare from_cc string and "Taiwan" in Chinese (Big5)
+ * from_cc will generate when user login bbs system, and set from FROMD
+ *
+ * Return true when user is from Taiwan IP, false when outside Taiwan
+ */
+
+#ifdef POST_COMMENT_WITH_TAIWAN_IP_ONLY
+bool
+is_taiwan_ip()
+{
+    // TODO: move this string to string table, eg: common.h
+    return !strncmp("�O�W", from_cc, STRLEN);
```

Comment:
Given from_cc is properly null terminated, use strcmp instead of strncmp.
If you don't trust the null character and want to be extremely safe, use sizeof(from_cc) instead of STRLEN, just in case the size may be changed in the future.

Reply:
Agree, using `sizeof(from_cc)` is better.